### PR TITLE
fix(eks): policy does not exist or is not attachable in China regions

### PR DESCRIPTION
**Pull request by pahud**
As ECR Public is not available in China regions and govcloud, `AmazonElasticContainerRegistryPublicReadOnly` is not available and should not be attached to the role. This PR implements a `RegionInfo` method to determine if ECR public is available in current region and conditionally attach `AmazonElasticContainerRegistryPublicReadOnly` policy to the role.

Closes https://github.com/aws/aws-cdk/issues/24743 https://github.com/aws/aws-cdk/issues/24808

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by pahud** [comment]
Exemption Request

File: packages/@aws-cdk/aws-eks/lib/private/service-available.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { Token } from '@aws-cdk/core';
+import { RegionInfo } from '@aws-cdk/region-info';
+
+export function EcrPublicAvailable(region: any): boolean {
+  if (region && !Token.isUnresolved(region)) {
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-eks/lib/private/service-available.ts:5]: Is there any handling we can do for when a token is being used? It seems like it's a fairly common use case and this fix would still result in an error for those users.

Comment (pahud) [reply packages/@aws-cdk/aws-eks/lib/private/service-available.ts:5]: Yes thank you for the heads up. I will fix this soon.

**Comment by pahud** [comment]
close in favor of https://github.com/aws/aws-cdk/pull/25170

**Pull request closed without merging**